mono/metadata/marshal.c

```diff
@@ -10013,6 +10017,76 @@ get_virtual_stelemref_wrapper (int kind)
 		mono_mb_emit_exception (mb, "ArrayTypeMismatchException", NULL);
 		break;
 	}
+
+	case STELEMREF_CLASS_SMALL_IDEPTH:
+		/*
```

Comment:
Yes, it's the same trick we do on the regular casting code. 
MonoClass::supertypes is never smaller than MONO_DEFAULT_SUPERTABLE_SIZE.

Second point, I haven't tried adding such fast path, but from the previous change that removed it from STELEMREF_CLASS, the hit count in not significant enough to justify it.